test/federated/test_train_config.py

```diff
@@ -435,9 +428,10 @@ def forward(self, x):
         print("Loss before training: {}".format(loss_before))
         print("Loss after training: {}".format(loss_after))
 
-    local_worker.ws.shutdown()
-    del local_worker
+    remote_proxy.close()
+    del remote_proxy
 
-    process_remote_worker.terminate()
+    time.sleep(0.1)
+    server.terminate()
```

Comment:
could you check whether the two lines:
del remote_proxy
and 
time.sleep(0.1)
are necessary?
The main issue with the websocket connections in the tests are connection requests after closing the connections.
These are mostly due to:
 - pointer objects that refer to remote objects (and are in charge of garbage collection) that are deleted at the end of the test

Reply:
will do

Reply:
I don't think we need them, getting rid of them.